src/ansys/additive/core/thermal_history.py

```diff
@@ -170,21 +169,55 @@ def __eq__(self, other: object) -> bool:
         return True
 
     @property
-    def geometry(self):
-        """Part geometry.
+    def id(self) -> str:
+        """User-provided ID for the simulation."""
+        return self._id
 
-        For more information, see the :class:`StlFile`
-        class or the :class:`BuildFile` class.
-        """
+    @id.setter
+    def id(self, value):
+        self._id = value
+
+    @property
+    def machine(self) -> AdditiveMachine:
+        """Machine parameters."""
+        return self._machine
+
+    @machine.setter
+    def machine(self, value):
+        self._machine = value
+
+    @property
+    def material(self) -> AdditiveMaterial:
+        """Material parameters."""
+        return self._material
+
+    @material.setter
+    def material(self, value):
+        self._material = value
+
+    @property
+    def geometry(self) -> StlFile | BuildFile:
+        """Part geometry."""
         return self._geometry
 
     @geometry.setter
     def geometry(self, value):
         """Set geometry."""
         if not isinstance(value, (StlFile, BuildFile)):
-            raise TypeError("ThermalHistoryInput.geometry must be an StlFile of BuildFile")
+            raise TypeError("Geometry must be an StlFile of BuildFile")
```

Comment:
```suggestion
            raise TypeError("Geometry must be an StlFile of BuildFile.")
```
Messages should conclude with punctuation. Can the message be changed to use the extension or name (given that I don't think Build is an extension but generic type): 
"Geometry must be an STL file or build file."